src/utils/getCourseScore.js

```diff
@@ -1,40 +1,33 @@
 const getCourseScore = ({ subjects: coursePrimary, faculty: courseFaculty }, { primary, lingual }) => {
-  let maxPrimaryScore = 0
-  let maxLingualScore = 0
-
-  let maxLingualSubject = {}
-
-  primary.forEach((subject) => {
-    if (subject.max.score > maxPrimaryScore && subject.active && coursePrimary.indexOf(subject.name) !== -1) {
-      maxPrimaryScore = subject.max.score
-    }
-  })
-
-  lingual.forEach((subject) => {
-    if (subject.max.score > maxLingualScore && subject.active) {
-      maxLingualSubject = subject
-      maxLingualScore = subject.max.score
+  const primaryArray = primary.filter(subject => coursePrimary.indexOf(subject.name) !== -1 && subject.active)
```

Comment:
This and subjectPicker are actually pretty similar. I'm sure there is a cleaver way to avoid duplication. ( ͡° ͜ʖ ͡°)